linalg/ode.cpp

```diff
@@ -362,6 +364,12 @@ AdamsBashforthSolver::AdamsBashforthSolver(int s_, const double *a_)
    {
       RKsolver = new RK4Solver();
    }
+
+#ifdef MFEM_USE_MPI
+   print = mfem::Mpi::IsInitialized() ? mfem::Mpi::Root() : true;
```

Comment:
What does `mfem::MPI::Root()` return if `mfem::Mpi::IsInitialized == false`? Asking to gain an understanding for this conditional.

Reply:
I believe it will result in a runtime MPI error. `Mpi::Root()` has to make a call to the MPI API, which will error if `MPI_Init` has not been called yet.

Reply:
Also, I don't think these need to be qualified with `mfem::`. It should be fine to use `Mpi::IsInitialized` and `Mpi::Root` since this is within the `mfem` namespace.

Reply:
Ah I see that now in the `Mpi` singleton. In my opinion, the correct behavior if this line is reached and `MPI_Init` has not yet been called is an error is generated. If there's agreement on that, I recommend the conditional be removed and any error handling for the case when the user has forgotten to call `Mpi::Init` be left for implementation in the `Mpi` singleton.

Reply:
It's perfectly valid to compile MFEM with MPI, but not call `MPI_Init` (as long as none of the parallel functionality is used). This is what happens in all of the serial examples when MFEM is configured to use MPI.

Reply:
That makes sense.  What do the other MFEM classes that are called from both serial and parallel examples do when they want to print to the terminal?

Reply:
I think most instances have similar logic. (Yes, there is probably a good opportunity for refactoring, or adding something like `ParPrint` that handles this automatically, but I would recommend doing that separately from this PR).